src/com/dataiku/hive/udf/maps/UDAFMapGroupMin.java

```diff
@@ -0,0 +1,391 @@
+                mapAppend(agg.map, o);
+            }
+        }
+
+        @Override
+        public Object terminatePartial(AggregationBuffer ab) throws HiveException {
+            MapBuffer agg = (MapBuffer) ab;
+            return Maps.newHashMap(agg.map);
+        }
+
+        @Override
+        public void merge(AggregationBuffer ab, Object p) throws HiveException {
+            MapBuffer agg = (MapBuffer) ab;
+            @SuppressWarnings("unchecked")
+            Map<Object, Object> obj = (Map<Object, Object>) this.originalDataOI.getMap(p);
+            mapAppend(agg.map, obj);
+        }
+
+        @Override
+        public Object terminate(AggregationBuffer ab)  throws HiveException {
+            MapBuffer agg = (MapBuffer) ab;
+            return Maps.newHashMap(agg.map);
+        }
+    }
+
+    public static class MapGroupMinLongEvaluator extends GenericUDAFEvaluator {
+        private MapObjectInspector originalDataOI;
+        private LongObjectInspector valueOI;
+        private StringObjectInspector keyOI;
+
+
+        @Override
+        public ObjectInspector init(Mode m, ObjectInspector[] parameters) throws HiveException {
+            super.init(m, parameters);
+
+            originalDataOI = (MapObjectInspector) parameters[0];
+            keyOI = (StringObjectInspector) originalDataOI.getMapKeyObjectInspector();
+            valueOI = (LongObjectInspector) originalDataOI.getMapValueObjectInspector();
+            return ObjectInspectorFactory.getStandardMapObjectInspector(PrimitiveObjectInspectorFactory.javaStringObjectInspector,
+                        PrimitiveObjectInspectorFactory.javaLongObjectInspector);
+
+        }
+
+        static class MapBuffer implements AggregationBuffer {
+            Map<String, Long> map = new HashMap<String, Long>();
+        }
+
+        @Override
+        public void reset(AggregationBuffer ab) throws HiveException {
+            ((MapBuffer) ab).map.clear();
+        }
+
+        @Override
+        public AggregationBuffer getNewAggregationBuffer() throws HiveException {
+            return new MapBuffer();
+        }
+
+        protected void mapAppend(Map<String, Long> m, Map<Object, Object> from)  {
+            if (from == null) {
+                return;
+            }
+            for(Map.Entry<Object, Object> entry : from.entrySet()) {
+                Object okey = entry.getKey();
+                Object ovalue = entry.getValue();
+                if (okey == null || ovalue == null) continue;
+                String key = keyOI.getPrimitiveJavaObject(entry.getKey());
+                Long value = valueOI.get(entry.getValue());
+                if (m.containsKey(key)) {
+                    Long currValue = m.get(key);
+                    if (currValue == null || value < currValue) {
+                        m.put(key, value);
+                    }
+                } else {
+                    m.put(key, value);
+                }
```

Comment:
Minor remark, but this could also be expressed as:
```java
if(!m.containsKey(key) || value < m.get(key) || m.get(key) == null){
     m.put(key, value)
}
```